library/packages/src/lib/packages/repository_product.rb

```diff
@@ -0,0 +1,17 @@
+module Packages
+  class RepositoryProduct
+    attr_reader :name, :version, :arch, :status, :category
+
+    def initialize(name:, version:, arch:, status:, category:)
+      @arch = arch.to_sym
+      @name = name
+      @status = status.to_sym
+      @category = category.to_sym
+      @version = version
+    end
+
+    def ==(other)
+      arch == other.arch && name == other.name && version == other.version
```

Comment:
Maybe we should also store the `vendor` attribute and compare it as well.

It's very unlikely to have the very same product (in the same version) from also from a different vendor, but...